spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/Sample.scala

```diff
@@ -16,132 +16,306 @@
-    require(this.isInstanceOf[TensorSample[T]], "Deprecated method, Only support TensorSample.")
-    this.asInstanceOf[TensorSample[T]].labelTensor
+  def label()(implicit ev: TensorNumeric[T]): Tensor[T] = {
+    throw new UnsupportedOperationException("Sample.label(): unimplemented deprecated method")
   }
 
+  /**
+   * Set data of feature and label.
+   * @param featureData
+   * @param labelData
+   * @param featureSize
+   * @param labelSize
+   * @return
+   */
   @deprecated("Old interface", "0.2.0")
   def set(
         featureData: Array[T],
         labelData: Array[T],
         featureSize: Array[Int],
-        labelSize: Array[Int]): Sample[T] = {
-    require(this.isInstanceOf[TensorSample[T]], "Deprecated method, Only support TensorSample.")
-    val sample = this.asInstanceOf[TensorSample[T]]
-    sample.featureTensor.set(Storage[T](featureData), 1, featureSize)
-    sample.labelTensor.set(Storage[T](labelData), 1, labelSize)
-    sample
-  }
+        labelSize: Array[Int])(implicit ev: TensorNumeric[T]): Sample[T]
+
+  /**
+   * Get feature sizes
+   * @return feature sizes
+   */
+  def getFeatureSize(): Array[Array[Int]]
+
+
+  /**
+   * Get label sizes
+   * @return label sizes
+   */
+  def getLabelSize(): Array[Array[Int]]
+
+  /**
+   * Get data
+   * @return data
+   */
+  def getData(): Array[T]
 }
 
+
 /**
- * A kind of sample. Feature is a tensor, and label is a tensor too.
- * @param featureTensor feature tensor
- * @param labelTensor label tensor
- * @tparam T numeric type
+ * A kind of sample who use only one array
  */
-class TensorSample[T: ClassTag](
-      val featureTensor: Tensor[T],
-      val labelTensor: Tensor[T]) extends Sample[T] {
+private[bigdl] class ArraySample[T: ClassTag](
+      private val data: Array[T],
+      private val featureSize: Array[Array[Int]],
+      private val labelSize: Array[Array[Int]]) extends Sample[T] {
 
-  /**
-   * The length of first dimension
-   * @return The length of first dimension
-   */
-  override def featureLength(): Int = {
-    featureTensor.size(1)
+  override def getData(): Array[T] = data
+
+  override def featureLength(index: Int): Int = {
+    require(null != featureSize, "featureSize is empty")
+    featureSize(index)(0)
   }
 
-  override def copy(other: Sample[T]): this.type = {
-    require(other.isInstanceOf[TensorSample[T]], "Sample.copy: sample type not match.")
-    val s = other.asInstanceOf[TensorSample[T]]
-    featureTensor.resizeAs(s.featureTensor).copy(s.featureTensor)
-    labelTensor.resizeAs(s.labelTensor).copy(s.labelTensor)
-    this
+  override def labelLength(index: Int): Int = {
+    if (null == labelSize) {
+      0
+    } else {
+      labelSize(index)(0)
+    }
+  }
+
+  override def getFeatureSize(): Array[Array[Int]] = {
+    featureSize
   }
 
-  def numFeature(): Int = 1
+  override def getLabelSize(): Array[Array[Int]] = {
+    if (null != labelSize) {
+      labelSize
+    } else {
+      Array(Array(0))
```

Comment:
But Array(Array(0)) is not immutable, 
User may change the content:
```
scala> val a = Array(Array(0))
a: Array[Array[Int]] = Array(Array(0))

scala> a(0)
res0: Array[Int] = Array(0)

scala> a(0) = Array(2)

scala> a
res2: Array[Array[Int]] = Array(Array(2))
```
Return an new array is much safe.